### PR TITLE
Feat: added converters for rcnn models

### PR DESCRIPTION
**Pull request by niobeus**
Added ConvTranspose converters. Small fixes for RoiAlign and Squeeze nodes

### CODE REVIEW

File: onnx2torch/node_converters/roialign.py [hunk 1]
```diff
@@ -31,23 +33,57 @@ def __init__(
...
+            args = [input_tensor, rois, batch_indices]
+            return _RoiAlignExportToOnnx.set_output_and_apply(output, *args)
+
+        return output
+
+class _RoiAlignExportToOnnx(CustomExportToOnnx):  # pylint: disable=abstract-method
+
+    @staticmethod
+    def symbolic(graph: torch_C.Graph, *args) -> torch_C.Value:
+        return graph.op('RoiAlign', *args, outputs=1)
```
Comment (ivkalgin) [inline onnx2torch/node_converters/roialign.py:85]: You forgot about attributes
```python
output_height : int (default is 1)
output_width : int (default is 1)
sampling_ratio : int (default is 0)
spatial_scale : float (default is 1.0)
```

File: onnx2torch/node_converters/roialign.py [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
 from torch import nn
 from torchvision.ops import roi_align
 
 from onnx2torch.node_converters.registry import add_converter
 from onnx2torch.onnx_graph import OnnxGraph
 from onnx2torch.onnx_node import OnnxNode
-from onnx2torch.utils.common import OnnxToTorchModule
 from onnx2torch.utils.common import OperationConverterResult
 from onnx2torch.utils.common import onnx_mapping_from_node
+from onnx2torch.utils.custom_export_to_onnx import CustomExportToOnnx, OnnxToTorchModuleWithCustomExport
```
Comment (ivkalgin) [inline onnx2torch/node_converters/roialign.py:15]: Should be split on 2 lines

File: onnx2torch/node_converters/roialign.py [hunk 3]
```diff
@@ -31,23 +33,57 @@ def __init__(
...
             rois: torch.Tensor,
             batch_indices: torch.Tensor,
+            output_size: Tuple[int, int],
+            spatial_scale: float,
+            sampling_ratio: int,
     ) -> torch.Tensor:
-        batched_rois = torch.concat([batch_indices.unsqueeze(1).to(rois.dtype), rois], dim=1)
+
+        batch_indices = batch_indices.unsqueeze(1).to(rois.dtype)
+
```
Comment (ivkalgin) [inline onnx2torch/node_converters/roialign.py:47]: please remove empty line

File: onnx2torch/node_converters/conv.py [hunk 4]
```diff
@@ -11,14 +11,19 @@
...
+    ('ConvTranspose', 1): nn.ConvTranspose1d,
+    ('ConvTranspose', 2): nn.ConvTranspose2d,
+    ('ConvTranspose', 3): nn.ConvTranspose3d,
 }
 
 
 @add_converter(operation_type='Conv', version=1)
 @add_converter(operation_type='Conv', version=11)
+@add_converter(operation_type='ConvTranspose', version=1)
+@add_converter(operation_type='ConvTranspose', version=11)
```
Comment (ivkalgin) [inline onnx2torch/node_converters/conv.py:26]: Please, update tests.

File: onnx2torch/node_converters/squeeze.py [hunk 5]
```diff
@@ -30,10 +30,13 @@ def __init__(self, axes: Optional[List[int]] = None):
             axes = sorted(axes, reverse=True)
 
         self.axes = axes
+        # We specify a forward in order to avoid 'if'-statements in the computational graph
+        self.forward = self._default_forward if not axes else self._specific_forward
 
-    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
-        if not self.axes:
-            return torch.squeeze(input_tensor)
+    def _default_forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
```
Comment (ivkalgin) [inline onnx2torch/node_converters/squeeze.py:36]: `_default_forward` -> `_torch_squeeze_forward`
`_specific_forward` -> `_onnx_squeeze_forward`

File: onnx2torch/node_converters/roialign.py (hunk 2 above)
Comment (niobeus) [reply onnx2torch/node_converters/roialign.py:15]: Done

File: onnx2torch/node_converters/roialign.py (hunk 3 above)
Comment (niobeus) [reply onnx2torch/node_converters/roialign.py:47]: Done

File: onnx2torch/node_converters/squeeze.py (hunk 5 above)
Comment (niobeus) [reply onnx2torch/node_converters/squeeze.py]: Done

File: onnx2torch/node_converters/conv.py (hunk 4 above)
Comment (niobeus) [reply onnx2torch/node_converters/conv.py:26]: Done

File: onnx2torch/node_converters/roialign.py (hunk 1 above)
Comment (niobeus) [reply onnx2torch/node_converters/roialign.py:91]: Done

Comment (irkjero) [reply onnx2torch/node_converters/roialign.py:91]: PEP 8
Immediately before a comma, semicolon, or colon:
# Correct:
if x == 4: print(x, y); x, y = y, x
# Wrong:
if x == 4 : print(x , y) ; x , y = y , x

File: onnx2torch/node_converters/roialign.py [hunk 6]
```diff
@@ -23,30 +26,69 @@ def __init__(
...
-            aligned=False,
         )
+        if torch.onnx.is_in_onnx_export():
+            args = [input_tensor, rois, batch_indices]
+            kwargs = {
+                'output_height' : self._output_height, 
+                'output_width' : self._output_width, 
+                'sampling_ratio' : self._sampling_ratio, 
+                'spatial_scale' : self._spatial_scale
+            }
```
Comment (ivkalgin) [inline onnx2torch/node_converters/roialign.py:82]: 1) It is useless kwargs, which will be silently ignored
2) names of attributes must contain type suffix

File: onnx2torch/node_converters/roialign.py (hunk 1 above)
Comment (niobeus) [reply onnx2torch/node_converters/roialign.py]: Done

File: onnx2torch/node_converters/roialign.py (hunk 6 above)
Comment (niobeus) [reply onnx2torch/node_converters/roialign.py]: Kwargs are removed

File: onnx2torch/node_converters/squeeze.py [hunk 7]
```diff
@@ -31,16 +30,25 @@ def __init__(self, axes: Optional[List[int]] = None):
...
         result = input_tensor
-        for axes_id in self.axes:
+        for axes_id in axes:
             result = torch.squeeze(result, dim=axes_id)
 
         return result
 
+    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
+        output = self._do_forward(input_tensor, self.axes)
+        if torch.onnx.is_in_onnx_export() and opset_version == 13:
```
Comment (ivkalgin) [inline onnx2torch/node_converters/squeeze.py:46]: opset_version >= 13

File: onnx2torch/node_converters/squeeze.py [hunk 8]
```diff
@@ -89,6 +104,7 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint:
     if len(node.input_values) == 2:
         try:
             axes = get_const_value(node.input_values[1], graph)
+            axes = axes.tolist()
```
Comment (ivkalgin) [inline onnx2torch/node_converters/squeeze.py:107]: merge with develop

File: onnx2torch/node_converters/squeeze.py (hunk 7 above)
Comment (niobeus) [reply onnx2torch/node_converters/squeeze.py:46]: Ok

File: onnx2torch/node_converters/squeeze.py (hunk 8 above)
Comment (niobeus) [reply onnx2torch/node_converters/squeeze.py]: Done

File: onnx2torch/node_converters/squeeze.py [hunk 9]
```diff
@@ -9,44 +9,60 @@
...
+        if not self.is_empty_axes(axes):
             axes = sorted(axes, reverse=True)
 
         self.axes = axes
 
-    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
-        if not self.axes:
+    @staticmethod
+    def is_empty_axes(axes: List[int]) -> bool:
+        return axes is None
```
Comment (ivkalgin) [inline onnx2torch/node_converters/squeeze.py:35]: in this case `is_empty_axes` function is not necessary

```python
if not self.axes:
    ...
```
will be enough

Comment (niobeus) [reply onnx2torch/node_converters/squeeze.py]: Done

File: onnx2torch/node_converters/roialign.py [hunk 10]
```diff
@@ -1,18 +1,21 @@
...
 from onnx2torch.onnx_node import OnnxNode
-from onnx2torch.utils.common import OnnxToTorchModule
 from onnx2torch.utils.common import OperationConverterResult
 from onnx2torch.utils.common import onnx_mapping_from_node
+from onnx2torch.utils.custom_export_to_onnx import CustomExportToOnnx
+from onnx2torch.utils.custom_export_to_onnx import OnnxToTorchModuleWithCustomExport
 
-
-class OnnxRoiAlign(nn.Module, OnnxToTorchModule):
+class OnnxRoiAlign(nn.Module, OnnxToTorchModuleWithCustomExport):
```
[the remark this answers is not in the document]
Comment (niobeus) [reply onnx2torch/node_converters/roialign.py:19]: Sorry... Pylint doesn't show me messages about it. I will add empty lines

File: onnx2torch/node_converters/roialign.py [hunk 11]
```diff
@@ -23,31 +26,82 @@ def __init__(
...
+                self._output_height,
+                self._output_width,
+                self._sampling_ratio,
+                self._spatial_scale
+            ]
+            return _RoiAlignExportToOnnx.set_output_and_apply(output, *args)
+
+        return output
+
+class _RoiAlignExportToOnnx(CustomExportToOnnx):  # pylint: disable=abstract-method
```
[the remark this answers is not in the document]
Comment (niobeus) [reply onnx2torch/node_converters/roialign.py:91]: Sorry... Pylint doesn't show me messages about it. I will add empty lines

File: onnx2torch/node_converters/squeeze.py [hunk 12]
```diff
@@ -9,44 +9,56 @@
...
-from onnx2torch.utils.common import OnnxToTorchModule
 from onnx2torch.utils.common import OperationConverterResult
 from onnx2torch.utils.common import get_const_value
 from onnx2torch.utils.common import onnx_mapping_from_node
 from onnx2torch.utils.custom_export_to_onnx import CustomExportToOnnx
 from onnx2torch.utils.custom_export_to_onnx import OnnxToTorchModuleWithCustomExport
 
-
-class OnnxSqueezeStaticAxes(nn.Module, OnnxToTorchModule):
+class OnnxSqueezeStaticAxes(nn.Module, OnnxToTorchModuleWithCustomExport):
```
[the remark this answers is not in the document]
Comment (niobeus) [reply onnx2torch/node_converters/squeeze.py:23]: Added

**Review by ivkalgin — APPROVED** [review]

**Pull request merged**